src/internal_modules/roc_fec/writer.h

```diff
@@ -77,7 +77,7 @@ class Writer : public packet::IWriter, public core::NonCopyable<> {
     //! @remarks
     //!  - writes the given source packet to the output writer
     //!  - generates repair packets and also writes them to the output writer
-    virtual void write(const packet::PacketPtr&);
+    virtual ROC_ATTR_NODISCARD status::StatusCode write(const packet::PacketPtr&);
```

Comment:
Do we need the attribute in derived classes, or only in the interface?

Reply:
In derived classes too, otherwise nodisard error won't be triggered.

Nodisard is triggered:

```cpp
void foo(const packet::PacketPtr& pp, packet::IWriter& writer) {
   writer.write(pp);
}
```

Nodisard isn't triggered:

```cpp
struct FooWriter : public packet::IWriter {
   virtual status::StatusCode write(const packet::PacketPtr& pp) {
   }
};

void foo(const packet::PacketPtr& pp, FooWriter& writer) {
   writer.write(pp);
}
```